cooltools/insulation.py

```diff
@@ -152,14 +170,26 @@ def calculate_insulation_score(
         A table containing the insulation scores of the genomic bins and
         the insulating boundary strengths.
     """
-    if chromosomes is None:
-        chromosomes = clr.chromnames
+
+    if view_df is None:
+        view_df = make_cooler_view(clr)
+    else:
+        # Make sure view_df is a proper viewframe
+        try:
+            _ = is_compatible_viewframe(
+                view_df,
+                clr,
+                check_sorting=True,
+                raise_errors=True,
+            )
+        except Exception as e:
+            raise ValueError("view_df is not a valid viewframe or incompatible") from e
 
     bin_size = clr.info["bin-size"]
     ignore_diags = (
         ignore_diags
```

Comment:
i think @Phlya was in favor of "simple" `if then else` instead of this one-liners - others agreed - should we modify ?

I also think a simple `if then else` is probably cleaner , unless a one-liner `if then else` really fits in 1 line